apps/s_client.c

```diff
@@ -1742,7 +1762,11 @@ int s_client_main(int argc, char **argv)
     if (fallback_scsv)
         SSL_set_mode(con, SSL_MODE_SEND_FALLBACK_SCSV);
 
-    if (servername != NULL) {
+    if (!noservername && (servername != NULL || dane_tlsa_domain == NULL)) {
```

Comment:
The guard here is correct. The implicit SNI name is not needed with `dane_tlsa_domain`, otherwise explicit from `-servername` or implicit without `-noservername` is used.